src/Evolution/DiscontinuousGalerkin/Limiters/SimpleWenoImpl.hpp

```diff
@@ -0,0 +1,123 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <boost/functional/hash.hpp>  // IWYU pragma: keep
+#include <cstddef>
+#include <limits>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/Element.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/Mesh.hpp"
+#include "Evolution/DiscontinuousGalerkin/Limiters/WenoGridHelpers.hpp"
+#include "Evolution/DiscontinuousGalerkin/Limiters/WenoHelpers.hpp"
+#include "Evolution/DiscontinuousGalerkin/Limiters/WenoOscillationIndicator.hpp"
+#include "NumericalAlgorithms/Interpolation/RegularGridInterpolant.hpp"
+#include "NumericalAlgorithms/LinearOperators/MeanValue.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Limiters {
+namespace Weno_detail {
+
+// Compute the Simple WENO solution for one tensor component
+//
+// This interface is intended for use in limiters that check the troubled-cell
+// indicator independently for each tensor component. These limiters generally
+// need to limit only a subset of the tensor components.
+template <typename Tag, size_t VolumeDim, typename PackagedData>
+void simple_weno_impl(
+    const gsl::not_null<std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>,
```

Comment:
You can include MortarHelpers.hpp and use `dg::MortarMap`, it was added in #2037

Reply:
I agree this is a heavy type, and I've previously thought of making a type alias for it. I would prefer to do so in a different PR, because it should really be done uniformly in all the limiters, so I think it makes more sense to do it all at once (after this rebase is over).

Reply:
I'm fine with postponing the type correction until after the rest of the refactor is completed.

Reply:
I'm  all for a type alias, but let's not have the WENO code depend on mortar things... Making limiters depend on mortars is the kind of stuff @kidder and I have been trying to get rid of

Reply:
Don't fret, I would not have wanted to use that particular type alias 🙂 

However, since we are now seeing this kind of structure pop in a few places, I might suggest adding (somewhere at the `domain` level?) a type alias `NeighborMap` for an `unordered_map` of blah that is indexed by `pair<Direction, ElementId>`. This would be a nice complement to the `DirectionMap` we already have that is indexed by `Direction` only.

Reply:
:+1:

I like the name `NeigborMap`. There used to be (back when I was young) a `DirectionalElementId` that was basically the `pair<Direction, ElementId>` so another name (that I like less than `NeighborMap`) is `DirectionalElementMap`. We would probably want to have it be implemented the same way as `DirectionMap`, a class that inherits off of the `std::unordered_map<...>`. Thoughts?

Reply:
👍